Silo: Patch for HDF5 1.14

kwryankrattiger opened the pull request.

kwryankrattiger committed:
Silo: Patch for HDF5 1.14

eschnett commented:
Silo extends some internal data structures of HDF5. These data structures can change between HDF5 version, and this patch updates Silo to support HDF5 1.14. This looks fine to me.

kwryankrattiger commented:
This fixes part of the issues I found, I get a bit further with this patch. I am going to keep digging on these issues and try to get all of the patches in the same PR here.

eschnett commented:
@kwryankrattiger Do you want to make this a "draft PR" until it's ready to be reviewed?

kwryankrattiger commented:
@eschnett I think this is ready to review. I opened another issue in Silo for my other errors which seem to be unrelated to HDF5 compatibility but rather with Silo itself.

https://github.com/LLNL/Silo/issues/309

eschnett commented:
I looked at a diff of the file `H5FDsilo.c` between the Silo release 4.11 and the current Silo HEAD. There are other changes to HDF5 data structures as well https://gist.github.com/eschnett/54d593e2d5a88e4a011b83a08ecfea52: There are two more lines with `NULL` inserted, and there is the value `512` inserted in another place. I don't know whether this is significant, but it might be good to incorporate these changes as well.

kwryankrattiger commented:
Ah yes, I see those aren't in the 4.11 release yet either, but they are patched already in the `hdf5-113.patch` which was introduced before HDF5 1.13.2 compatibility.

@markcmiller86 The patches that went in for the other parts should also probably be back ported to 4.11RC LLNL/Silo#260

markcmiller86 commented:
> @markcmiller86 The patches that went in for the other parts should also probably be back ported to 4.11RC [LLNL/Silo#260](https://github.com/LLNL/Silo/pull/260)

Yep. Did that immediately after I merged your patch, https://github.com/LLNL/Silo/pull/308

kwryankrattiger committed:
Silo: Patch addition fixes to HDF5 drv

Update Data and Vis SDK stack to build VisIt with HDF5 1.14. This is not the
deployment spec for E4S, which is going to drop VisIt due to using old
Python and VTK, but an effort to keep testing DaV SDK packages with the
latest possible software.

kwryankrattiger commented:
@eschnett I got the other patch needed for the HDF5 driver in Silo 4.11 to work. I also update the CI stack for the DaV to force building VisIt with HDF5 1.14.

eschnett commented:
@kwryankrattiger CI fails because there is a syntax error in your patch.

kwryankrattiger commented:
@spackbot fix style

eschnett commented:
@kwryankrattiger There is an actual syntax error: `var/spack/repos/builtin/packages/silo/package.py:88: [E999] SyntaxError: invalid syntax`. You will need to correct this manually.
